File: benchmark/baselines/typescript-strict/capabilities.ts
import type { Registration, Result } from "./domain.js";
export interface RegistrationStore {
  find(eventId: number, personId: number): Registration | undefined;
  save(registration: Registration): Result<void, "store_failed">;
}
export interface Clock { now(): number }
export interface Notification { send(message: string): void }

File: benchmark/baselines/rust/capabilities.rs
use crate::domain::Registration;
pub trait RegistrationStore {
    fn find(&self, event_id: i64, person_id: i64) -> Option<Registration>;
    fn save(&mut self, registration: Registration) -> Result<(), ()>;
}
pub trait Clock { fn now(&mut self) -> i64; }
pub trait Notification { fn send(&mut self, message: &str); }
